tests/test_poetry.py

```diff
@@ -0,0 +1,142 @@
+"""Unit and integration tests for recipifying Poetry projects."""
+
+import filecmp
+from pathlib import Path
+
+import pytest
+
+from grayskull.__main__ import generate_recipes_from_list, init_parser
+from grayskull.strategy.py_toml import (
+    InvalidVersion,
+    add_poetry_metadata,
+    encode_poetry_version,
+    get_all_toml_info,
+    get_caret_ceiling,
+    get_tilde_ceiling,
+    parse_version,
+)
+
+
+def test_parse_version():
+    assert parse_version("0") == {"major": 0, "minor": None, "patch": None}
+    assert parse_version("1") == {"major": 1, "minor": None, "patch": None}
+    assert parse_version("1.2") == {"major": 1, "minor": 2, "patch": None}
+    assert parse_version("1.2.3") == {"major": 1, "minor": 2, "patch": 3}
+
+    with pytest.raises(InvalidVersion):
+        parse_version("asdf")
+    with pytest.raises(InvalidVersion):
+        parse_version("")
+    with pytest.raises(InvalidVersion):
+        parse_version(".")
```

Comment:
can we divide this test in two parametrized tests please?
Something like
```suggestion
@pytest.mark.pararetrize(
    "version, major, minor, patch",
    [
        ("0", 0, None, None),
        ("1", 1, None, None),
        ("1.2", 1, 2, None),
        ("1.2.3", 1, 2, 3),
    ]
)
def test_parse_version_success(version, major, minor, patch):
    assert parse_version(version) == {"major": major, "minor": minor, "patch": patch}


@pytest.mark.parametrize("invalid_version", ["asdf", "", "."])
def test_parse_version_failure(invalid_version):
    with pytest.raises(InvalidVersion):
        parse_version(invalid_version)
```

Reply:
Done in latest revision.